Cython/Utility/ModuleSetupCode.c

```diff
@@ -852,11 +841,22 @@ static CYTHON_INLINE void *__Pyx_PyModule_GetState(PyObject *op)
 }
 #endif
 
+// The "TryGetSlot" variants may return NULL on static types with the Limited API on earlier versions
+// so should be used for optimization rather than where a result is required.
 #define __Pyx_PyObject_GetSlot(obj, name, func_ctype)  __Pyx_PyType_GetSlot(Py_TYPE((PyObject *) obj), name, func_ctype)
+#define __Pyx_PyObject_TryGetSlot(obj, name, func_ctype) __Pyx_PyType_TryGetSlot(Py_TYPE(obj), name, func_ctype)
+#define __Pyx_PyObject_TryGetSubSlot(obj, sub, name, func_ctype) __Pyx_PyType_TryGetSubSlot(Py_TYPE(obj), sub, name, func_ctype)
```

Comment:
Most of this is just copied from https://github.com/cython/cython/pull/6169

Reply:
I added a few comments in that PR so that it can be merged first.